Gems/GradientSignal/Code/Source/Components/PerlinGradientComponent.cpp

```diff
@@ -148,10 +148,12 @@ namespace GradientSignal
 
     void PerlinGradientComponent::Deactivate()
     {
-        m_perlinImprovedNoise.reset();
         GradientRequestBus::Handler::BusDisconnect();
         PerlinGradientRequestBus::Handler::BusDisconnect();
         GradientTransformNotificationBus::Handler::BusDisconnect();
+
+        AZStd::unique_lock<decltype(m_transformMutex)> lock(m_transformMutex);
```

Comment:
The decltype isn't needed.
Class Template argument deduction will be able to deduce the unique_lock template type based on the constructor parametes.

```suggestion
        AZStd::unique_lock lock(m_transformMutex);
```

Reply:
nice, thanks for the suggestion! Copied it from another place just in case if this is intentional